javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -386,7 +402,9 @@ public final void write(ClientModel model, JavaFile javaFile) {
 
                     // getter
                     generateGetterJavadoc(classBlock, model, property);
-
+                    if (targetProperty.getIsReadOnly()) {
+                        addJsonGetter(classBlock, settings, targetProperty.getSerializedName());
+                    }
```

Comment:
You might not want to add `JsonGetter` or `JsonSetter` at this level. This part of the code provides getter/setter for flattened properties (i.e., `data` under `properties`), which is not valid at this level of class.

Reply:
Sample code here.
https://github.com/Azure/autorest.java/blob/main/vanilla-tests/src/main/java/fixtures/discriminatorflattening/clientflatten/models/LoadBalancer.java

Reply:
Hmm, this is a tricky scenario than. The aim for this change was to better align serialization with getters and setters to reduce/prevent Jackson from needing to `setAccessible` private Fields (effectively breaking encapsulation). But the flattened inner properties don't generate a setter, nor are they part of a constructor.

This will need further investigation.